sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/NotificationHook/NotificationHook.cs

```diff
@@ -43,14 +53,20 @@ internal NotificationHook()
         public string Description { get; set; }
 
         /// <summary> Optional field which enables a customized redirect, such as for troubleshooting notes. </summary>
-        public string ExternalLink { get; set; }
+        public Uri ExternalLink { get; set; }
+
+        /// <summary>
+        /// Used by CodeGen during serialization.
+        /// </summary>
+        [CodeGenMember("ExternalLink")]
+        internal string InternalExternalLink => ExternalLink?.AbsoluteUri;
```

Comment:
Simply updating `ExternalLink` from `string` to `Uri` unfortunately won't work. The MA service make optional strings empty by default. After creating a hook, if `ExternalLink` has not been specified, which is allowed, the service will return an empty string when we call GetHook, and the generated code will try to create an `Uri` from the empty string and will throw.

So we have an `InternalExternalLink` (part of serialization) which is kept as a string. We manually convert it to the exposed `Uri` member, `ExternalLink`.

Just realized that `InternalExternal` is a funny combination.

Reply:
lol yeah. the name is kind of weird, but I couldn't come up with a better one :P